src/core/runtime.hpp

```diff
@@ -301,8 +298,9 @@ inline bool tick(SHWire *wire, SHDuration now) {
     wire->coro->resume();
 #endif
 
-#ifdef SH_USE_TSAN
-    __tsan_switch_to_fiber(curr, 0);
+#ifdef TRACY_FIBERS
+    TracyFiberLeave;
```

Comment:
This isn't correct I think, if you're switching back to another wire you should call Enter but with it's name instead (Try Step/Do/etc., the profiler just says invalid zone, left twice, etc.)

Tracy doesn't keep a stack. Probably the easiest way to fix this is to pass an (optional) current context to switching functions (tick, prepare) so you can switch back to the calling 'fiber' after returning. Or keep a thread_local stack somewhere (just for the profiler)

Reply:
hmm, not sure if that is an issue, should not be the case

Reply:
But maybe this is exactly what's happening with threads related shards... 💡

Reply:
Threadlocal wont help sadly. Flow might. I'll think about it.

Reply:
Thing is profiler says there never was a thread.. not twice. But might be related indeed. But not sure about the wire thing though. Wire pointer won't ever change cos it's decided when call is called. No matter when we resume.